src/vs/workbench/contrib/search/common/searchModel.ts

```diff
@@ -806,9 +807,9 @@ export class SearchResult extends Disposable {
 				}
 
 				if (elem instanceof FileMatch) {
-					elem.parent().replace(elem);
+					await elem.parent().replace(elem);
 				} else if (elem instanceof Match) {
-					elem.parent().replace(elem);
+					await elem.parent().replace(elem);
```

Comment:
added await here also because I think we should be waiting for the replaceQ to be done too?